test/e2e/e2e.go

```diff
@@ -1871,3 +1872,185 @@ func getNodePodCIDR(nodeName string) (string, error) {
+		northDBFileName string        = "ovnnb_db.db"
+		southDBFileName string        = "ovnsb_db.db"
+		dirDB           string        = "/etc/ovn/"
+		waitingPeriod   time.Duration = 90 * time.Second // polling timeout
+		ovnWorkerNode   string        = "ovn-worker"
+		ovnWorkerNode2  string        = "ovn-worker2"
+		haModeMinDb     int           = 0
+		haModeMaxDb     int           = 2
+	)
+	var (
+		containerToDeleteFrom = databasePodNorthContainer
+	)
+	f := framework.NewDefaultFramework(svcname)
+
+	deletePod := func(f *framework.Framework, namespace string, podName string) {
+		podClient := f.ClientSet.CoreV1().Pods(namespace)
+		err := podClient.Delete(context.Background(), podName, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "should delete ovnkube-node pod")
+	}
+
+	deleteAndRestartDB := func(db_pod_num int, DBFileNamesToDelete []string, containerToDeleteFrom string) {
+		var (
+			containerFlag = fmt.Sprintf("-c=%s", containerToDeleteFrom)
+			db_pod_name   = fmt.Sprintf("%s-%d", databasePodPrefix, db_pod_num)
+			errStr        string
+		)
+		checkDbFilesExistence := func(db_pod_num int) {
+			out, _ := framework.RunKubectl(ovnNs, "exec", db_pod_name, containerFlag, "--", "ls", dirDB)
+			out = strings.TrimSuffix(strings.ReplaceAll(out, "\n", ","), ",")
+			framework.Logf("current db files:%s", out)
+			ginkgo.By("current db files:" + out)
+			errStr = ""
+			if !strings.Contains(out, northDBFileName) {
+				errStr += " " + northDBFileName
+			}
+			if !strings.Contains(out, southDBFileName) {
+				errStr += " " + southDBFileName
+			}
+			if errStr != "" {
+				errStr = "missing db:" + errStr
+				framework.Failf(errStr)
+			}
+		}
+		checkDbFilesExistence(db_pod_num)
+		//delete db file(s)
+		for _, DBFileNameToDelete := range DBFileNamesToDelete {
+			framework.Logf("deleting db file %s", DBFileNameToDelete)
+			framework.RunKubectl(ovnNs, "exec", db_pod_name, containerFlag, "--", "rm", dirDB+DBFileNameToDelete)
+			//check db files- make sure its deleted
+			out, _ := framework.RunKubectl(ovnNs, "exec", db_pod_name, containerFlag, "--", "ls", dirDB)
+			out = strings.TrimSuffix(strings.ReplaceAll(out, "\n", ","), ",")
+			if out == "" {
+				framework.Logf("all db files deleted")
+			} else {
+				framework.Logf("cur db files:%s", out)
+			}
+			if strings.Contains(out, DBFileNameToDelete) {
+				errStr = "db not deleted:" + DBFileNameToDelete
+				framework.Failf(errStr)
+			}
+		}
+
+		//delete the db-pod in order to emulate the cluster/pod restart
+		framework.Logf("deleting db pod:%s", db_pod_name)
+		deletePod(f, ovnNs, db_pod_name)
+
+		//wait (let the new db pod to be ready)
+		framework.Logf("wait %v", waitingPeriod)
+		time.Sleep(waitingPeriod)
+
+		//check db files existence
+		checkDbFilesExistence(db_pod_num)
+	}
+	singlePodConnectivityTest := func(f *framework.Framework, podName string) {
+		framework.Logf("Running container which tries to connect to 8.8.8.8 in a loop")
```

Comment:
Rather than connecting to 8.8.8.8 (which is external and can therefore fail), this can just check that we can access the API server.

Reply:
Done, thanks.